wallet/wallet.go

```diff
@@ -421,16 +463,46 @@ func (w *Wallet) syncWithChain() error {
 				return err
 			}
 
+			// Check to see if this header's timestamp has surpassed
+			// our birthday. If we are in recovery mode and the
+			// check passes, we will add this block to our list of
+			// blocks to scan for recovered addresses.
+			timestamp := header.Timestamp
+			if isRecovery && timestamp.After(w.Manager.Birthday()) {
+				recoveryMgr.AddToBlockBatch(
+					hash, height, timestamp,
+				)
+			}
+
 			err = w.Manager.SetSyncedTo(ns, &waddrmgr.BlockStamp{
 				Hash:      *hash,
 				Height:    height,
-				Timestamp: header.Timestamp,
+				Timestamp: timestamp,
 			})
 			if err != nil {
 				tx.Rollback()
 				return err
 			}
 
+			// If we are in recovery mode, attempt a recovery on
+			// blocks that have been added to the recovery manager's
+			// block batch thus far. If block batch is empty, this
+			// will be a NOP.
+			if isRecovery && height%2000 == 0 {
```

Comment:
If this is the same `variable =2000` as used other places, it should probably be extracted into a constant.